File: app/src/main/java/com/cjstudio/sosestrada/MainActivity.java
package com.cjstudio.sosestrada;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

public class MainActivity extends AppCompatActivity {

    private CardView cardMotorista, cardPrestador;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        cardMotorista = findViewById(R.id.cardMotorista);
        cardPrestador = findViewById(R.id.cardPrestador);

        // Motorista → LoginMotoristaActivity
        cardMotorista.setOnClickListener(v -> {
            salvarPerfil("motorista");
            startActivity(new Intent(MainActivity.this, LoginMotoristaActivity.class));
        });

        // Prestador → LoginPrestadorActivity
        cardPrestador.setOnClickListener(v -> {
            salvarPerfil("prestador");
            startActivity(new Intent(MainActivity.this, LoginPrestadorActivity.class));
        });
    }

    private void salvarPerfil(String tipo) {
        SharedPreferences preferences = getSharedPreferences("SOSEstradaPrefs", MODE_PRIVATE);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString("tipoPerfil", tipo);
        editor.apply();
    }
}

File: app/src/main/java/com/cjstudio/sosestrada/MainActivity.kt
package com.cjstudio.sosestrada

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.cjstudio.sosestrada.databinding.ActivityMainBinding

// Tela inicial do app de usuários: escolhe entre motorista e prestador.
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.cardMotorista.setOnClickListener {
            startActivity(Intent(this, LoginMotoristaActivity::class.java))
        }
        binding.cardPrestador.setOnClickListener {
            startActivity(Intent(this, LoginPrestadorActivity::class.java))
        }
    }
}
